### ISSUE TITLE
What is value of walkoff compared to other workflow frameworks/ bpm frameworks?

### ISSUE THREAD

**Issue by StephenOTT** [nsacyber/WALKOFF#199]
Hey

reviewing walkoff usage and trying to understand the value in using of walkoff when compared to other workflow frameworks such as BPM tooling/frameworks.  Any docs on this?

**Comment by StephenOTT** [comment]
Anyone from the project able to provide any context?

Thanks

**Comment by shadowbq** [comment]
(Not a core member) I would think that BPM tooling/frameworks normally are a higher level concept than a pure orchestration framework like WALKOFF. I would consider WALKOFF more akin to Phantom than to APPIAN. But that being said the Security Orchestration community is looking at BMPN for sharing defined workflows. https://static1.squarespace.com/static/5a94b67ff93fd440f0516297/t/5bd495d0e2c4835088fc6fcd/1540658677973/Day+1+BS2+K7K8+IACD+Sharable+Workflows+IC+October+2018.pdf

**Comment by StephenOTT** [comment]
@shadowbq can define what you mean by "pure orchestration"?

Thanks for the .pdf share!

**Comment by shadowbq** [comment]
@StephenOTT All of them bleed into each other. Its hard to really tell if it fits your needs. General BPM from my understand covers Orchestration processes, Collaboration processes, and Choreography processes.. unless you get into the nitty. I've used WALKOFF to set firewall rules, turnoff lights, and send email, etc.. 

I've not used it to model Buisness Process with people, data pools/lakes, task polls, and participant usage over multiple processes, etc.. those I would assume live better elsewhere.

**Comment by iadgovuser11** [comment]
Sorry for the delay in responding.  When designing WALKOFF we've tried to focus on automating processes between devices and capabilities such as through APIs or platforms.  Many times when we're building workflows the core question becomes "what buttons are you clicking over and over?".   Our core use case is cybersecurity but WALKOFF doesn't restrict you;  you can automate just about anything.   I hope this helps and let me know if you have any other questions.

**Comment by StephenOTT** [comment]
@iadgovuser11 thanks for followup! So the "purpose" of walkoff was always clear.  But what I am trying to find is the "~unique?" qualities that walkoff is solving compared to other tooling for generic process/workflow capabilities.
A few examples that i am just making up could be: it is python based, it uses a specific db or does not require a db, it meets a specific standard or sec standard, it has a N sized footprint that was for specific requirements, etc.

The buttons clicking over and over again is common BPMN and RPA style flows. (forget about Human tasks in BPMN).  So i am looking at what walkoff was trying to solve because there was a gap in the existing tooling.

Thanks!

**Comment by iadgovuser11** [comment]
We've tried to focus as much as possible on ease of use whether that be deploying it, or creating workflows.  For example the interfaces feature we've created is because we've found that its just as important to interpret the automation as it is to actually automate.  At the end of the day that ease of use is what is going to get WALKOFF to be used.   

We also wanted to create a system flexible enough to handle as many use cases as possible.  There are a lot of different places that can use automation.  We wanted to create an accessible platform to bring those automated capabilities to people with the ability to share apps and workflows among the community.    We found that making WALKOFF open source is the way to accomplish that.  

Thanks!  Let me know if you have any other questions.
